src/main/kotlin/org/example/chitchatserver/domain/chat/usecase/QueryChatHistoriesService.kt

```diff
@@ -0,0 +1,43 @@
+package org.example.chitchatserver.domain.chat.usecase
+
+import org.example.chitchatserver.common.exception.NotFoundException
+import org.example.chitchatserver.domain.chat.persistence.repository.ChatRepository
+import org.example.chitchatserver.domain.chat.persistence.repository.RoomRepository
+import org.example.chitchatserver.domain.chat.presentation.dto.response.ChatHistoryResponse
+import org.example.chitchatserver.domain.chat.presentation.dto.response.QueryChatHistoriesResponse
+import org.example.chitchatserver.domain.user.facade.UserFacade
+import org.springframework.data.domain.PageRequest
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import java.util.UUID
+
+@Service
+class QueryChatHistoriesService(
+    private val chatRepository: ChatRepository,
+    private val roomRepository: RoomRepository,
+    private val userFacade: UserFacade,
+) {
+
+    fun execute(roomId: UUID, page: Int): Mono<QueryChatHistoriesResponse> = roomRepository
+        .findById(roomId)
+        .switchIfEmpty(Mono.error(NotFoundException("NotFound")))
+        .flatMap { room ->
+            val name = room.topic
+            userFacade.getCurrentUserId()
+                .flatMapMany { currentUserId ->
+                    chatRepository.findAllByRoomIdOrderBySendAtDesc(room.id, PageRequest.of(page, 50))
+                        .map { chat ->
+                            ChatHistoryResponse(
+                                chat.nickname,
+                                chat.profileImageUrl,
+                                chat.content,
+                                chat.userId == currentUserId,
+                                chat.type,
+                                chat.sendAt,
+                            )
+                        }
+                }
+                .collectList()
+                .flatMap { chats -> Mono.just(QueryChatHistoriesResponse(name, chats)) }
+        }
+}
```

Comment:
이거 currentUserId랑 room 정보 가져오는거를
Mono.zip()으로 묶어서
<img width="369" alt="image" src="https://github.com/letsgochitchat/chitchat-server/assets/52336493/f1761bd0-fc94-4c84-9df9-045d3522f366">
대충 이런식으로 처리하면 flatMap지옥 없앨 수 있을 거같은데 어떰

Reply:
시도를 해봤는데 분리하려면 flatMapMany를 건드릴수밖에 없는데 저거 건드리면 왠지 모르겠는데 최종 반환이 null이 되어서 결과값이 아예 없어져버림... 그래서 실패함

Reply:
```suggestion
       Mono.zip(
            roomRepository.findById(roomId)
                .switchIfEmpty(Mono.error(NotFoundException("Room Not Found"))),
            userFacade.getCurrentUserId()
        ).flatMap {
            val room = it.t1
            val currentUserId = it.t2
            chatRepository.findAllByRoomIdOrderBySendAtDesc(roomId, PageRequest.of(page, 50))
                .map { chat ->
                    ChatHistoryResponse(
                        chat.nickname,
                        chat.profileImageUrl,
                        chat.content,
                        chat.userId == currentUserId,
                        chat.type,
                        chat.sendAt!!,
                    )
                }.collectList()
                .defaultIfEmpty(listOf())
                .map { chatHistories ->
                    QueryChatHistoriesResponse(room.topic, chatHistories)
                }
        }
```
대충 이렇게 하니까 작동하는거같은데 확인해보셈
+ main브렌치에 jwt header검사하는쪽에 문제가 있는거같아서 내가 채팅 개발하는 브랜치에는 고쳐놨는데 여긴 안고쳐져서 테스트할라면 9번 브랜치에서 jwt error라고 커밋해놓은거 참고해서 고친다음에 해야될듯

Reply:
200은 되는데 그 response가 안옴

Reply:
차피 테스트할때 permitall로 바꾸고 해서 크게 상관 없엇음